Cue/app/actions/library.js

```diff
@@ -17,4 +19,89 @@ async function loadLibrary(): PromiseAction {
   };
 }
 
-module.exports = { loadLibrary };
+function deleteDeck(uuid: string) : Action {
+  return {
+    type: 'DECK_DELETED',
+    uuid
+  }
+}
+
+function createDeck(name: string) : Action {
+  let deck = {
+    name,
+    uuid: uuidV4(),
+    public: false,
+    cards: [],
+    owner: CueApi.userId,
+    accession: 'private',
+  }
+  return {
+    type: 'DECK_CREATED',
+    deck,
+  }
+}
+
+function editDeck(change) : Action {
+  return {
+    type: 'DECK_EDITED',
+    change
+  }
+}
+
+//attempts to sync each deck once, returns list of failed syncs
+function syncLibrary(localChanges): ThunkAction {
+  return (dispatch, getState) => {
+    let failedSyncs = []
+    let promises = []
+    localChanges.forEach(change =>{
+      promises.push(dispatch(syncDeck(change))
+        .catch(e => {
+          failedSyncs.push(change)
+        }))
+    })
+    return Promise.all(promises).then(() => {
```

Comment:
Actually on second look, we already catch any promises which reject. So this won't be an issue?